sympy/functions/special/error_functions.py

```diff
@@ -190,20 +190,31 @@ def _eval_conjugate(self):
         return self.func(self.args[0].conjugate())
 
     def _eval_is_real(self):
-        return self.args[0].is_extended_real
+        if self.args[0].is_extended_real is True:
+            return True
+        # There are cases where erf(z) becomes a real number
+        # even if z is a complex number
+
+    def _eval_is_imaginary(self):
+        t = self.args[0].extract_multiplicatively(I)
+        if t is not None and t.is_real is True and t.is_zero is False:
+            return True
```

Comment:
Is this not equivalent to `if self.args[0].is_imaginary`?

Reply:
Indeed... this is terrible.